pandas/tests/frame/indexing/test_indexing.py

```diff
@@ -860,6 +860,15 @@ def test_fancy_getitem_slice_mixed(self, float_frame, float_string_frame):
 
         assert (float_frame["C"] == 4).all()
 
+    def test_setitem_slice_position(self):
```

Comment:
@jbrockmendel  if you can also test (df[:4]) for example (can just target master is ok)